kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemUtil.java

```diff
@@ -227,7 +238,7 @@ static boolean cleanlyDelete(FileSystem fs, Path root, Path path) {
           if (stats == null || stats.length == 0) {
             // dir is empty and should be removed
             LOG.debug("Deleting empty path {}", current);
-            deleted = fs.delete(current, true) || deleted;
+            deleted = (useTrash ? Trash.moveToAppropriateTrash(fs, current, fs.getConf()) : fs.delete(current, true)) || deleted;
```

Comment:
I don't think moving to trash is appropriate here. We know that a child directory of the pending-trashed directory is already in trash (assuming deleted = true as of line 231), so moving a now-empty parent directory would just create a timestamp-suffixed duplicate empty directory in trash, at each level of the path hierarchy. Reverting back to the original code here should be completely safe and a lot cleaner.